### ISSUE TITLE
env CENTRIFUGO_PORT not handled

### ISSUE THREAD

**Issue by grachevko** [centrifugal/centrifugo#223]
Environemnt CENTRIFUGO_PORT don't work to set port. 
I guess it becase 8000 is predefined for cli. 

https://github.com/centrifugal/centrifugo/blob/eb8d6b9586eaf90bf21059344a14a6f519bb4419/libcentrifugo/centrifugo/main.go#L213

This is a bug or feature?

**Comment by FZambia** [comment]
Hello, it's expected - port can not be set via environment at moment. To let config var to be set via environment it must be explicitly used while viper set up, so diff like this will do the work:

```
diff --git a/libcentrifugo/centrifugo/main.go b/libcentrifugo/centrifugo/main.go
index 3a9ede7..93efb62 100644
--- a/libcentrifugo/centrifugo/main.go
+++ b/libcentrifugo/centrifugo/main.go
@@ -222,7 +222,7 @@ func ConfigureServer(setter config.Setter) error {
        }
 
        bindEnvs := []string{
-               "web",
+               "web", "port",
        }
 
        for _, env := range bindEnvs {
```

Do you need this or possibility to set port via command-line is enough for your case?

**Comment by grachevko** [comment]
Our infrastructure build on Docker and Docker swarm, and we have 4 different usage of centrifugo: local, test, rc and prod. 
I built centrifugo docker image for solve this issue https://github.com/centrifugal/centrifugo-docker/issues/4, and wanted to use one point to configure some default options like port. But it's not a big problem, just not much convenient.

It will be great if we can use official centrifugo image for all cases out of the box without extending it.

**Comment by FZambia** [comment]
@grachevko see bf8655914ef94aaa4b2579d943b64fc63e7b9b08

Regarding to Docker Swarm secrets - I had not enough time to think on it and can't decide at moment whether we should add `entrypoint.sh` or port recursive directory handling to `viper-lite` lib (fork of Viper that Centrifugo uses internally).

**Comment by grachevko** [comment]
@FZambia thanks. 

In my opinion handle `*_FILE` on `viper` will more convenient than entypoint, because entrypoint will require to duplicate handled environments to sh script.

**Comment by FZambia** [comment]
> entrypoint will require to duplicate handled environments to sh script.

Not sure I understand this point. I thought it's possible to not duplicate config env names in `entrypoint` - just look at directory with secrets, then add `CENTRIFUGO_` prefix to a name of each file inside dir and then set as env var with file content as value. Am I missing something?

**Comment by grachevko** [comment]
1. I never see implementation like that
2. What if not all secrets related to centrifugo and one of secrets will have collision with centrifugo envs?
3. You restrict developer in naming of secret files.

So the way `if <env> not found try check <env>_FILE and get from it` more flexible.

**Comment by FZambia** [comment]
As far as I understand (never worked with Swarm though) when giving service permission to secrets only those secrets will be mounted in container.

Personally I am more for viper approach too, just because this will allow to keep docker image the same.

**Comment by grachevko** [comment]
Do not limit the flexibility of loading secrets from files only for Docker swarm needs.

**Comment by FZambia** [comment]
Released in [v1.8.0](https://github.com/centrifugal/centrifugo/releases/tag/v1.8.0)
